src/Data/Singletons/Single/Data.hs

```diff
@@ -30,7 +31,8 @@ singDataD (DataDecl _nd name tvbs ctors derivings) = do
   let tvbNames = map extractTvbName tvbs
   k <- promoteType (foldType (DConT name) (map DVarT tvbNames))
   ctors' <- mapM (singCtor a) ctors
-
+  ctorFixities <- singFixityDeclarations
```

Comment:
That this works is a delicate balance that should probably be spelled out.

If we are singling a previously defined datatype (whose definition is retrieved by reification), then the `qReifyFixity` calls underneath `singFixityDeclarations` work. If we are singling a datatype being defined right now (e.g., in a call to `singletons`) then this works because we will also encounter the fixity declaration and single it separately. But wait: does the `DsMonad` machinery succeed for `qReifyFixity` in this case? Would we then get two singled fixity declarations?